canonical-web-and-design/maas-ui
# feat(ui): build add partition form in machine storage

## Done

- Added `actions` parameter to normalised storage devices, which will house the actions that a storage device can perform (similar to how `machine.actions` works)
- Added util for determining whether a storage device can be partitioned
- Added `AddPartition` form

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to the storage tab of a "Ready" or "Allocated" machine
- Check that physical disks in "Available disks and partitions" have an action dropdown that can open the "Add partition" form
- Check that adding a formatted partition adds it to the "Filesystems" table
- Check that adding an unformatted partition adds it to the "Available disks and partitions" table

## Fixes

Fixes canonical-web-and-design/MAAS-squad#2164

## Screenshot

![Screenshot_2020-11-25 fun-mammal maas storage bolla MAAS](https://user-images.githubusercontent.com/25733845/100172332-d33ff400-2f13-11eb-9807-7017b56f7a6b.png)
